app/src/main/java/com/example/concordia_campus_guide/Models/Building.java

```diff
@@ -72,7 +75,7 @@
     public Building(){}
     public Building(Double[] centerCoordinates, List<String> availableFloors, float width, float height, float bearing,
                     String campus, String buildingCode, String Building_Long_Name, String address,
-                    List<String> departments, List<String> services, List<List<List<Double>>> coordinates) {
+                    List<String> departments, List<String> services, ListOfCoordinates cornerCoordinates) {
         super(centerCoordinates);
```

Comment:
I think we can start thinking about using the Builder design pattern over here maybe if it is applicable.

Reply:
feel free to do so, maybe open a ticket for improvement, I don't think it should be done in this pr